theano/tensor/nnet/abstract_conv2d.py

```diff
@@ -0,0 +1,741 @@
+"""
+FIXME
+"""
+
+__docformat__ = "restructuredtext en"
+
+import logging
+
+import numpy
+
+import theano
+from theano.tensor import (as_tensor_variable, blas, get_scalar_constant_value,
+                           patternbroadcast, NotScalarConstantError)
+from theano.tensor import TensorType
+from theano.gof import Apply, Op
+from theano.gof import local_optimizer
+
+from theano.sandbox.cuda import register_opt as register_gpu
+from theano.tensor.opt import register_specialize_device
+
+
+### Gpu related optimization (to be moved in sandbox/cuda)
+from theano.sandbox.cuda.basic_ops import (
+    as_cuda_ndarray_variable,
+    gpu_contiguous, gpu_from_host, host_from_gpu,
+    GpuFromHost, HostFromGpu
+    )
+from theano.sandbox.cuda.type import CudaNdarrayType
+from theano.sandbox.cuda.dnn import dnn_available, dnn_conv, dnn_gradweight, dnn_gradinput
+from theano.sandbox.cuda.blas import GpuCorrMM, GpuCorrMM_gradWeights, GpuCorrMM_gradInputs
+from theano.sandbox.cuda.opt import values_eq_approx_high_tol
+
+
+## Cpu implementation
+from theano.tensor.nnet import conv2d as cpu_conv2d, ConvOp
+from theano.tensor.nnet.ConvGrad3D import convGrad3D
+from theano.tensor.nnet.ConvTransp3D import convTransp3D
+
+_logger = logging.getLogger("theano.tensor.nnet.conv2d")
+
+
+def conv2d(inputs,
+           filters,
+           inputs_shape=None,
+           filters_shape=None,
+           batch_size=None,
```

Comment:
What is that for? It's not explained below, and the batch size is included in the input shape according to the documentation.